### PR TITLE
Downgrade fitting error

### PR DESCRIPTION
**Pull request by jvavrek**
Return 0 error estimate and warn instead of crashing in `Fitter.calc_area_and_unc()`

### CODE REVIEW

File: tests/fitting_test.py [hunk 1]
```diff
@@ -334,10 +334,11 @@ def test_component_areas(self, sim_high_stat):
             covariance = np.array(fitter.result.covariance)
         else:
             covariance = np.array(fitter.result.covar)
-        # We can at least check that it properly errors, then skip the rest of the tests
+        # We can at least check that it properly warns, then skip the rest of the tests
         if not covariance.sum():
```
Comment (markbandstra) [inline tests/fitting_test.py:338]: Please make sure this test is guaranteed to be evaluated.

File: becquerel/core/fitting.py [hunk 2]
```diff
@@ -892,7 +893,11 @@ def _calc_area(param_vec, **kwargs):
         else:
             covariance = np.array(self.result.covar)
         if not covariance.sum():
```
Comment (markbandstra) [inline becquerel/core/fitting.py:895]: Why not use `if np.isclose(covariance.sum(), 0):`?

Comment (markbandstra) [reply becquerel/core/fitting.py:895]: Or is it returning `None`?

File: tests/fitting_test.py (hunk 1 above)
Comment (markbandstra) [reply tests/fitting_test.py:338]: Also see the `if not covariance.sum()` comment above.

File: becquerel/core/fitting.py (hunk 2 above)
Comment (jvavrek) [reply becquerel/core/fitting.py]: Yes, it was returning None, and then np.array(None).sum() was behaving strangely. Cleaned up in the latest commit!

**Review by markbandstra — APPROVED** [review]

**Pull request merged**